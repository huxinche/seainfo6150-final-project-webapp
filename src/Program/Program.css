.paragraph{
    margin-bottom: 70px;
}

.paragraph p{
    margin: 30px auto;
    text-align: left;
}

.paragraph h2{
    color: #003558;
    font-family: "TiemposText",Georgia,serif;
    margin-bottom: 0.6em;
    margin-top: 1em;
}

.btn-back{
    transition: 0.2s ease;
    background-color: transparent;
    border: 2px solid #138387;
    border-radius: 30px;
    color: #138387;
    font-size: 13px;
    padding: .6em 2em;
}

.btn-back a:visited{
    color: inherit;
}

.paragraph img{
   width:  100%;
}

@media (min-width: 768px){
    .paragraph h2 {
        font-size: 2.08591em;
    }

    .btn-back{
        padding: 0.75em 2.25em;
    }
}