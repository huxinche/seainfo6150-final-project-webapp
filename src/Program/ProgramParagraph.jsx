import React from 'react';
import PropTypes from "prop-types";

const ProgramParagraph = (props) => {
    const { paragraph, center } = props
    return <div className={center ? "paragraph center" : "paragraph"}>
        <h2>{paragraph.title}</h2>
        <div dangerouslySetInnerHTML={{ __html: paragraph.text }}></div>
        <div className="center">
            {paragraph.image && <img  src={paragraph.image.url} alt="paragraph.title" />}
        </div>
    </div>
}

ProgramParagraph.propTypes = {
    paragraph: PropTypes.object.isRequired,
    center: PropTypes.bool,
}

export default ProgramParagraph;