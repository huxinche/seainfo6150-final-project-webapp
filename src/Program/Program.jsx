import React, { useEffect } from 'react';
import Banner from '../Components/Common/Banner/Banner';
import PropTypes from "prop-types";
import { Link } from 'react-router-dom';
import ProgramParagraph from './ProgramParagraph';

import './Program.css';


const Program = ({ program }) => {

    useEffect(()=>{
        window.scrollTo(0, 0)
    })

    return <> <Banner image={program['backgroud-image'].url}
        title={program.heading}
    />
        <div className="container">
            {program.paragraph1 && <ProgramParagraph paragraph={program.paragraph1} center></ProgramParagraph>}
            <hr />
            {program.paragraph2 && <ProgramParagraph paragraph={program.paragraph2}></ProgramParagraph>}
            {program.paragraph3 && <ProgramParagraph paragraph={program.paragraph3}></ProgramParagraph>}
            {program.paragraph4 && <ProgramParagraph paragraph={program.paragraph4}></ProgramParagraph>}
            <div style={{textAlign:'center'}}>
                <Link to="/findmyprogram" className="btn-back">Back to Programs</Link>
            </div>
            <br/>
            <br/>
        </div>

    </>
}

Program.propTypes = {
    program: PropTypes.object.isRequired
}


export default Program;