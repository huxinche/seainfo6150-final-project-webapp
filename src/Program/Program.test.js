import React from "react";
import Program from "./Program";

it("renders the Program component correctly", () => {
    window.scrollTo = jest.fn()
    let data = {

        "heading": "MS in Applied Data Analytics",
        "head-college": "B University Metropolitan College",
        "backgroud-image": {
            "url": "https://pim.northeastern.edu/sites/default/files/2020-06/Behrakis%20copy%202%20%281%29_0.jpg"
        },
        "description": "<p>The Master of Science in Applied Data Analytics is ideal for mid-career IT professionals or students with a computer science background who seek to train their focus on analytics. Students gain solid knowledge of data analytics and examine the presentation and applications of the latest industry tools and approaches within an academically rigorous framework.</p>",
        "college": "Metropolitan College",
        "degree": "Master's",
        "paragraph1": {
            "title": "Prepare for Critical Data Analytics Roles",
            "text": "<p>Available online and on campus, the Master of Science in Applied Data Analytics (MSADA) at Boston University’s Metropolitan College (MET) is a hands-on program that exposes you to various database systems, data mining tools, data visualization tools and packages, Python packages, R packages, and cloud services such as Amazon AWS, Google Cloud, and Mass Open Cloud. The knowledge of analytics tools combined with an understanding of data mining and machine learning approaches will enhance your ability to critically analyze real-world problems and understand the possibilities and limitations of analytics applications.</p>"
        },
        "paragraph2": {
            "title": "Advance Your Career with a Master’s in Applied Data Analytics",
            "text": "<p>With data analytics needs influencing every major industry—including healthcare, tech, finance, communication, entertainment, energy, transportation, government, and manufacturing, to name some—there is significant growth in specialized data science, data engineering, automation, AI, and machine learning areas. Yet the demand for skilled talent continues to outpace supply. QuantHub research confirms a shortfall of 250,000 data scientists in 2020, while McKinsey Global Institute anticipates as much as 12 percent annual growth in demand for graduates from data science programs over the next decade.</p><p>To harness the potential of this big data revolution, you need advanced techniques.</p><p>As a graduate of the MS in Applied Data Analytics program at BU MET, you will be able to demonstrate the ability to create powerful predictions through modeling and machine learning, and drive critical business decisions—skills needed to excel in a growing list of roles such as data scientist, economist, data analyst, business intelligence analyst, systems analyst, chief analytics officer, analytics manager, marketing analyst, business analyst, or financial analyst, among others.</p>"
        },
        "paragraph3": {
            "title": "Explore Careers in Data Analytics",
            "text": "<p>Use the Career Insights tool to explore jobs that are the right fit for you. Filter by career area and job title or by industry sector to explore employment demand and average salaries. Select “Learn More” for a downloadable career report, or “Explore Other Options” to find the BU MET degree or certificate program that will prepare you for the job you want.</p>",
            "image": {
                "url": "https://www.datasciencesociety.net/wp-content/uploads/2018/12/DataCamp-Data-Science-Background.png"
            }
        },
        "paragraph4": {
            "title": "Master the Tools to Excel in Applied Data Analytics",
            "text": "<p>Offered through BU MET’s Department of Computer Science, the Master of Science in Applied Data Analytics can set you apart by adding invaluable analytics expertise, skills, and projects to your résumé.</p><p>Ideal for mid-career IT professionals or students, BU MET’s Applied Data Analytics curriculum provides solid knowledge of data analytics and examines the presentation and applications of the latest industry tools and approaches within an academically rigorous framework. Emphasizing both data analytics and applied areas—including databases, applied machine learning, and large dataset processing methods—the Applied Data Analytics master’s curriculum provides a thorough immersion in concepts and techniques for organizing, cleaning, analyzing, and representing/visualizing large amounts of data.</p>"
        }

    }
    const { container } = render(<Program program={data} />);
    expect(container).toMatchSnapshot();
});
