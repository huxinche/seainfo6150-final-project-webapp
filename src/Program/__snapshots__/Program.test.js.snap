// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Program component correctly 1`] = `
<div>
   
  <div
    class="banner"
  >
    <div>
      <picture
        class="banner-img"
      >
        <img
          alt="MS in Applied Data Analytics"
          class="banner-img"
          src="https://pim.northeastern.edu/sites/default/files/2020-06/Behrakis%20copy%202%20%281%29_0.jpg"
        />
      </picture>
    </div>
    <div
      class="banner-content"
    >
      <div>
        <h1
          class="text-title"
        >
          MS in Applied Data Analytics
        </h1>
        <p
          class="text-title"
        />
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="paragraph center"
    >
      <h2>
        Prepare for Critical Data Analytics Roles
      </h2>
      <div>
        <p>
          Available online and on campus, the Master of Science in Applied Data Analytics (MSADA) at Boston University’s Metropolitan College (MET) is a hands-on program that exposes you to various database systems, data mining tools, data visualization tools and packages, Python packages, R packages, and cloud services such as Amazon AWS, Google Cloud, and Mass Open Cloud. The knowledge of analytics tools combined with an understanding of data mining and machine learning approaches will enhance your ability to critically analyze real-world problems and understand the possibilities and limitations of analytics applications.
        </p>
      </div>
      <div
        class="center"
      />
    </div>
    <hr />
    <div
      class="paragraph"
    >
      <h2>
        Advance Your Career with a Master’s in Applied Data Analytics
      </h2>
      <div>
        <p>
          With data analytics needs influencing every major industry—including healthcare, tech, finance, communication, entertainment, energy, transportation, government, and manufacturing, to name some—there is significant growth in specialized data science, data engineering, automation, AI, and machine learning areas. Yet the demand for skilled talent continues to outpace supply. QuantHub research confirms a shortfall of 250,000 data scientists in 2020, while McKinsey Global Institute anticipates as much as 12 percent annual growth in demand for graduates from data science programs over the next decade.
        </p>
        <p>
          To harness the potential of this big data revolution, you need advanced techniques.
        </p>
        <p>
          As a graduate of the MS in Applied Data Analytics program at BU MET, you will be able to demonstrate the ability to create powerful predictions through modeling and machine learning, and drive critical business decisions—skills needed to excel in a growing list of roles such as data scientist, economist, data analyst, business intelligence analyst, systems analyst, chief analytics officer, analytics manager, marketing analyst, business analyst, or financial analyst, among others.
        </p>
      </div>
      <div
        class="center"
      />
    </div>
    <div
      class="paragraph"
    >
      <h2>
        Explore Careers in Data Analytics
      </h2>
      <div>
        <p>
          Use the Career Insights tool to explore jobs that are the right fit for you. Filter by career area and job title or by industry sector to explore employment demand and average salaries. Select “Learn More” for a downloadable career report, or “Explore Other Options” to find the BU MET degree or certificate program that will prepare you for the job you want.
        </p>
      </div>
      <div
        class="center"
      >
        <img
          alt="paragraph.title"
          src="https://www.datasciencesociety.net/wp-content/uploads/2018/12/DataCamp-Data-Science-Background.png"
        />
      </div>
    </div>
    <div
      class="paragraph"
    >
      <h2>
        Master the Tools to Excel in Applied Data Analytics
      </h2>
      <div>
        <p>
          Offered through BU MET’s Department of Computer Science, the Master of Science in Applied Data Analytics can set you apart by adding invaluable analytics expertise, skills, and projects to your résumé.
        </p>
        <p>
          Ideal for mid-career IT professionals or students, BU MET’s Applied Data Analytics curriculum provides solid knowledge of data analytics and examines the presentation and applications of the latest industry tools and approaches within an academically rigorous framework. Emphasizing both data analytics and applied areas—including databases, applied machine learning, and large dataset processing methods—the Applied Data Analytics master’s curriculum provides a thorough immersion in concepts and techniques for organizing, cleaning, analyzing, and representing/visualizing large amounts of data.
        </p>
      </div>
      <div
        class="center"
      />
    </div>
    <div
      style="text-align: center;"
    >
      <a
        class="btn-back"
        href="/findmyprogram"
      >
        Back to Programs
      </a>
    </div>
    <br />
    <br />
  </div>
</div>
`;
